osfoffline/views/preferences.py

```diff
@@ -135,6 +156,7 @@ def create_tree_item_for_each_top_level_node(self):
 
             if node.id in user.guid_for_top_level_nodes_to_sync:
                 tree_item.setCheckState(self.PROJECT_SYNC_COLUMN, Qt.Checked)
+                self.checked_items.append(node.id)
```

Comment:
it needs to check if the node.id is already in the list. If you don't hit the update button, this list won't get assign as the guid_list and thus will append the node.id again make two lists different. Will fix this while I merge this pr.